src/abstractdatatable/reshape.jl

```diff
@@ -193,30 +188,17 @@ function unstack(dt::AbstractDataTable, rowkey::Int, colkey::Int, value::Int)
     # `rowkey` integer indicating which column to place along rows
     # `colkey` integer indicating which column to place along column headers
     # `value` integer indicating which column has values
-    refkeycol = NullableCategoricalArray(dt[rowkey])
```

Comment:
Thanks for the demo, that's very helpful to see. I'll add that example as a test case too.